tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -4957,17 +5027,27 @@ CudnnSupport::FusedConvolveRunnerFromDesc(
   TF_ASSIGN_OR_RETURN(auto op_graph,
                       GetCudnnFusedOperationGraph(
                           kind, input_type, bias_type, output_type, conv_scale,
-                          side_input_scale, input_descriptor, filter_descriptor,
-                          bias_descriptor, output_descriptor,
+                          side_input_scale, leakyrelu_alpha, input_descriptor,
+                          filter_descriptor, bias_descriptor, output_descriptor,
                           convolution_descriptor, activation_mode, cudnn));
 
   TF_ASSIGN_OR_RETURN(auto execution_plan,
                       RebuildExecutionPlan(cudnn, algorithm_desc, *op_graph));
 
+  // For historical reason, cudnn effectively does a Conv-Add-BiasAdd-Relu when
+  // the activation is Relu and it requires the "side_input" for the Add. For
+  // other activations, the "side_input" is no longer needed and we can do
+  // Conv-BiasAdd-Activation directly.
+  bool need_side_input = true;
+  if (activation_mode == dnn::ActivationMode::kRelu6 ||
+      activation_mode == dnn::ActivationMode::kElu ||
+      activation_mode == dnn::ActivationMode::kLeakyRelu) {
+    need_side_input = false;
+  }
```

Comment:
again, it seems like this logic should be flipped, in this case explicitly based on the comment.